subprojects/launcher/src/main/java/org/gradle/launcher/exec/BuildOutcomeReportingBuildActionRunner.java

```diff
@@ -57,8 +62,35 @@ public Result run(BuildAction action, BuildController buildController) {
 
         Result result = delegate.run(action, buildController);
 
+        Throwable failure = SingleMessageLogger.getDeprecationFailure();
+        if (failure != null) {
+            // Replace result if we have fail on warning
+            result = computeUpdatedResult(result, failure);
+        }
+
         buildLogger.logResult(result.getBuildFailure());
         new TaskExecutionStatisticsReporter(styledTextOutputFactory).buildFinished(taskStatisticsCollector.getStatistics());
         return result;
     }
+
+    private Result computeUpdatedResult(Result previousResult, Throwable deprecationFailure) {
+        if (previousResult.getBuildFailure() != null) {
+            // Enhance already reported failures
+            Throwable buildFailure = previousResult.getBuildFailure();
+            ArrayList<Throwable> newFailures;
+            if (buildFailure instanceof MultipleBuildFailures) {
+                MultipleBuildFailures multipleBuildFailures = (MultipleBuildFailures) buildFailure;
+                newFailures = Lists.newArrayListWithExpectedSize(multipleBuildFailures.getCauses().size() + 1);
+                newFailures.addAll(multipleBuildFailures.getCauses());
+            } else {
+                newFailures = Lists.newArrayListWithExpectedSize(2);
```

Comment:
minor: does it make sense to be this specific here?

Reply:
Well I have the info, so why not?

Reply:
It's just makes the code a little bit harder to change (i.e. if somebody adds a new entry they will have to recheck the declaration).
It's just a note, I don't mind either way :)